src/lib.rs

```diff
@@ -75,14 +56,21 @@ impl Calc {
                 prog.push(Instruction::Mul {})
             }
             AstNode::Number { value } => prog.push(Instruction::Push { value: value }),
+            AstNode::Print { rhs } => {
+                Self::to_bytecode(*rhs, prog);
+                prog.push(Instruction::Print {})
+            }
         }
     }
 
-    pub fn eval(instructions: &Vec<Instruction>) -> Result<u64, String> {
+    pub fn eval(instructions: &Vec<Instruction>) -> Result<Option<u64>, String> {
         let mut stack = vec![];
         for a in instructions {
             match a {
                 Instruction::Push { value } => stack.push(*value),
+                Instruction::Print {} => {
+                    println!("{}", stack.pop().expect("cannot pop from empty stack"))
```

Comment:
I thought for a while whether it was the right place for `println`, and I think it is.
REPL mode will have printing of its own.

Reply:
Looks ok to me.